### PR TITLE
Assign alliance through genesis block #99

### PR DESCRIPTION
**Pull request by hihiben**
##### Why:

To differentiate the ability of alliance and miner, a solution to assign these roles when initiating the blockchain is necessary.  
##### How:

By assigning the initial alliance through genesis block, we are able to assign the address as an alliance. This can also differentiate each individual blockchain at the beginning through a completely different genesis block, which means that the blockchain with pre-assigned alliance cannot be connected or mine for those who are not assigned. 
##### Usage:
-  The initial alliance address can be assigned in the configuration file (i.e. gcoin.conf). With the key `alliance` following by an address, the initial alliance address can be assigned, ex. `alliance= 18JHzqnhdgVbx9xghA4BXfe75Mp4bYHCKs`. If no alliance is assigned in genesis block, the mechanism remains the same as the former method (i.e. The first block miner will be the first alliance.)  
-  The addresses in the configuration file will be recorded in the genesis block by a form of `VOTE` transaction.  
-  By importing the corresponding key of the assigned address, user can assign the address as their default address through rpc `assignfixedaddress`.  
-  Mining start!  
##### Side Effects:

Since that the genesis block can be changed through the content of configuration file, the nonce of genesis block needs to be calculated in run-time, which means a little extra time is needed when starting the node.

### CODE REVIEW

File: src/chainparams.cpp [hunk 1]
```diff
@@ -161,6 +155,25 @@ class CMainParams : public CChainParams
             0
         };
     }
+
+    void AddAlliance(const std::string &addr) {
+        CMutableTransaction tx;
+        tx.vin.resize(1);
+        tx.vout.resize(1);
+        tx.type = VOTE;
+        const char* pszTimestamp = "OpenNet GCoin Project 2014.9 GCoin";
```
Comment (lbyyy) [inline src/chainparams.cpp:164]: 可以換一下了0.0

**Comment by zhouer** [comment]
是不是可以把設定初始 alliance 的多個 transactions 合成為一個 multisig address 的 transaction？

**Comment by hihiben** [comment]
這邊應該會跟著alliance的新增機制來做更動

File: src/chainparams.cpp (hunk 1 above)
Comment (hihiben) [reply src/chainparams.cpp:164]: 應該不需要換吧？

**Comment by zhouer** [comment]
pass

**Comment by lbyyy** [comment]
pass

**Pull request merged**